src/java/htsjdk/samtools/reference/ReferenceSequenceFileFactory.java

```diff
@@ -98,4 +100,57 @@ public static ReferenceSequenceFile getReferenceSequenceFile(final File file, fi
 
         throw new IllegalArgumentException("File is not a supported reference file type: " + file.getAbsolutePath());
     }
+
+    /**
+     * Attempts to determine the type of the reference file and return an instance
+     * of ReferenceSequenceFile that is appropriate to read it.  Sequence names
+     * will be truncated at first whitespace, if any.
+     *
+     * @param file the reference sequence file on disk
+     */
+    public static ReferenceSequenceFile getReferenceSequenceFile(final Path file) throws IOException {
```

Comment:
could we deduplicate the code here by having the old file constructors point to these constructors?